instruqt-tracks/terraform-cloud-aws-v2/03-centralize-deployments/check-workstation

```diff
@@ -0,0 +1,48 @@
+#!/bin/bash -l
+
+# Create /tmp/skip-check to disable this check
+if [ -f /tmp/skip-check ]; then
+    rm /tmp/skip-check
+    exit 0
+fi
+
+cd /root/hashicat-aws
+
+TOKEN=$(grep token /root/.terraform.d/credentials.tfrc.json | cut -d '"' -f4)
+
+# Check the token length
+[[ 91 -eq $(echo $TOKEN | wc -m) ]]
+EXIT_CODE=$?
+if [[ $EXIT_CODE -ne 0 ]]; then
+  fail-message "Oops, something's not right with your token. Check your /root/.terraform.d/credentials.tfrc.json file to be sure it has your token in it."
+fi
+
+# Does the user have an organization configured?
+grep -q YOURORGANIZATION /root/hashicat-aws/remote_backend.tf
+EXIT_CODE=$?
+if [[ $EXIT_CODE -eq 0 ]]; then
+  fail-message "Oops, it looks like you haven't added your organization name to the remote_backend.tf file yet."
+fi
+
+# Does the user have a workspace configured?
+grep -q $WORKSPACE /root/hashicat-aws/remote_backend.tf
+EXIT_CODE=$?
```

Comment:
One of these tests for a string, the other for a variable -- is that intentional?

Reply:
YOURORGANIZATION is a placeholder. If that string exists, then the user did not enter the correct info. We push that information to the remote_backend ourselves so it is impossible for the user to get it wrong.

$WORKSPACE is expressed as an environment variable is bashrc. If the variable contents is not expressed in the remote backend, then the match is not made and we exit. Again, we push that information to the remote_backend ourselves so it is nearly impossible for the user to get it wrong. 

The only way the user can get this wrong is by intentionally destroying the automated generation for this file. So, it is very intentional to add this sequence just to ensure that the user is not sabotaging the progress by manually changing the defaulted workspace and organization.